### DISCUSSION TITLE
the range of the values are different in Multidimensional Motif Discovery

### DISCUSSION THREAD

**Discussion by minw1012** [TDAmeritrade/stumpy#762] [Q&A]
<img width="618" alt="Screenshot 2022-12-12 at 4 05 36 PM" src="https://user-images.githubusercontent.com/14137107/207188432-9ed07fa2-7055-454f-8c72-4b1785afe0f9.png">

The above plot shown the two patterns from the two time series data are not in the same index.
if the range of values in T1 is [0,100] and the range of values in T1 is [200,500], will it influence the results in Multidimensional Motif Discovery algorithms?

Thanks

**Comment by seanlaw** [comment]
@minw1012 Can you please share the exact commands that you used to analyze the data and the commands to generate the plot? Without more detail, it is very difficult/impossible to help. There could be an infinite number of reasons why you are getting the results that you are showing.

**Comment by minw1012** [comment]
Thanks a lot for your quick reply.
<img width="923" alt="Screenshot 2022-12-12 at 5 03 52 PM" src="https://user-images.githubusercontent.com/14137107/207200917-a9bc2cea-ad2d-4f76-9812-2841e1202f4a.png">

<img width="583" alt="Screenshot 2022-12-12 at 5 04 03 PM" src="https://user-images.githubusercontent.com/14137107/207200934-b674d69d-99d6-4219-a26c-163290593e73.png">

**Reply by seanlaw** [reply]
> The above plot shown the two patterns from the two time series data are not in the same index.

I think that you may be misunderstanding what is being plotted. `motifs_idx[0]` gives you the (starting) index location of the motif when only one (out of two) dimensions are used to compute the matrix profile. However, this doesn't tell you any information about "which of the two dimensions" was used. Then, for `motifs_idx[1]`, this is the index location of the motif when both dimensions must be included in finding the "best" motif. 

> if the range of values in T1 is [0,100] and the range of values in T1 is [200,500], will it influence the results in Multidimensional Motif Discovery algorithms?

In this case, no, the range of the values should not matter because all distances are computed as z-normalized Euclidean distances (i.e., each subsequence is z-normalized first before comparing with another subsequence within the same time series). When `k = 2`, the motif distances from both time series are averaged and that is the definition of "multi-dimensional matrix profile".

**Reply by minw1012** [reply]
I got your idea. 
So the motifs_idx[1] is the multi-dimension motif.
I revised the codes. Could you double check the following figure?

Thanks a lot for your help.
<img width="922" alt="Screenshot 2022-12-12 at 6 14 44 PM" src="https://user-images.githubusercontent.com/14137107/207209480-828daa28-5eeb-4f35-9c53-82d0d98eaa07.png">

**Reply by seanlaw** [reply]
Yes, that looks about right
